### DISCUSSION TITLE
Status HM-LC-Bl1PBU-FM is working

### DISCUSSION THREAD

**Discussion by maxlin1** [SukramJ/aiohomematic#648] [Q&A]
Please answer all questions as good as possible.

Language:  german

Make sure you have read the documentation at https://github.com/danielperna84/custom_homematic#custom_homematic.

Make sure you have read the release notes at https://github.com/danielperna84/custom_homematic/releases. 

Make sure you are running the latest version of the custom_component (and Home Assistant) before reporting an issue.

ENSURE that your Backend (CCU/Homegear/...) is working as expected and connected devices are controllable.

Hi, I don't know if it's a bug or a feature, or if the error is due to RaspberryMatic.

**custom_component version (if applicable):**
1.18.1

**known last working custom_component version (if applicable):**
/

**Home Assistant version (if applicable):**
2022.10.1

**CCU version:**
- Model: (CCU3, Raspi4, NUC
- OS: RaspberryMatic
- Version: 

**Describe the bug**

I want to know when the screen is moving so I can use it in a rule to dim the light. I used to have this via openHAB here you can see when the screen is moving. Under homeassistant and RaspberryMatic I only get the status Open or Closed. Which is not always transmitted 100% correctly.

<img width="1070" alt="Bildschirmfoto 2022-10-10 um 19 48 48" src="https://user-images.githubusercontent.com/10881081/194925222-6a3848c8-c9ca-439b-ba7c-29e9856dd3d3.png">

**Additional context**

I am using a HM-LC-Bl1PBU-FM Homematic roller shutter actuator for controlling a screen with RaspberryMatic CCU
Current version: 3.65.11.20221005 https://github.com/jens-maus/RaspberryMatic

I want to know when the screen is moving so I can use it in a rule to dim the light. I used to have this via openHAB here you can see when the screen is moving. Under homeassistant and RaspberryMatic I only get the status Open or Closed. Which is not always transmitted 100% correctly. I've set a time in Homematic (36s until the screen is 100% down) but it sometimes shows 50% even when it's completely down. Sometimes the percentage value doesn't update. 

Does anyone have an idea why that is? is there a bug here?

**Comment by SukramJ** [comment]
I don't own such a device, but i cannot see an error in the code so far, but i think it's worth a discussion to see how other users handle this.
Basically, the status should show whether a cover is moving up or down.

**Reply by SukramJ** [reply]
> Language: german

Ich denke die weitere Diskussion kann dann auch auf deutsch weitergehen.

**Reply by maxlin1** [reply]
Sehr gerne, gibt es ein Log in dem ich so etwas sehen kann? unter OpenHub gibt es ein "Live Log"  bei homeassistant im "Verlauf jedoch sehe ich nur zu oder auf. 
<img width="907" alt="Bildschirmfoto 2022-10-10 um 20 29 19" src="https://user-images.githubusercontent.com/10881081/194930963-c31223c3-da2e-40f1-98fe-8b95dcacd326.png">

**Reply by SukramJ** [reply]
Nicht das ich wüsste. Was wird denn als Status der Cover Entität angezeigt während hoch oder runtergefahren wird?

**Reply by maxlin1** [reply]
Leider auch nur offen oder zu 
<img width="955" alt="Bildschirmfoto 2022-10-10 um 20 34 40" src="https://user-images.githubusercontent.com/10881081/194931601-fe6221ef-be56-4cf2-8fc7-4c89d96d4ded.png">

**Reply by SukramJ** [reply]
Das ist nicht ganz das was ich wissen wollte, bzw. die Auflösung ist zu grob
1. Gehe unter Entwicklerwerkzeuge/Zustände und suche die Entität. 
2. Steuere die Entität hoch/runter, ab besten in einem anderen Fenster/Gerät
3. Was steht in der Spalte Zustand zu Punkt 1?

**Reply by SukramJ** [reply]
Aktualisiere bitte HA und die custom_component auf die aktuelle Version.

**Reply by SukramJ** [reply]
Ansonsten mal mit dem [homematic-manager](https://github.com/hobbyquaker/homematic-manager) mit der CCU verbinden.
Dann das Gerät öffnen und unter Kanal 1 gibt es das Values Paramset. Die Frage ist dann, ob sich der Parameter DIRECTION bei Bewegung des Cover ändert.

**Comment by maxlin1** [comment]
Okay Update ist durch. 

hier zeigt er auch nur open oder closed an 
<img width="1064" alt="Bildschirmfoto 2022-10-10 um 20 53 18" src="https://user-images.githubusercontent.com/10881081/194934239-25454079-8b1c-487b-bae3-91a8fcec595f.png">

<img width="1291" alt="Bildschirmfoto 2022-10-10 um 20 57 27" src="https://user-images.githubusercontent.com/10881081/194934790-51a6bc78-3aec-4112-a766-f39811a51cbf.png">

**Reply by SukramJ** [reply]
Dann benutze mal den homematic-manager

**Comment by danielperna84** [comment]
Sollte das nicht der [WORKING](https://github.com/danielperna84/pydevccu/blob/30480d0050898eb3371e6d1b12c7857d70503dc2/pydevccu/paramset_descriptions/HM-LC-Bl1-FM.json#L178) parameter sein? Den sollte man sich doch mit unignore zurückholen können und dann einen `binary_sensor` erhalten der immer dann `True` ist, sobald sich was bewegt.

**Reply by SukramJ** [reply]
Mit WORKING kann man in HA eigentlich nichts anfangen. In DIRECTION sollte UP und DOWN stehen, und das kann man direkt in der Entität verwenden.

**Reply by maxlin1** [reply]
wie meinst du genau ?

**Reply by SukramJ** [reply]
Wer?

**Reply by maxlin1** [reply]
Wie kann ich WORKING als binary_sensor aktivieren ?

**Reply by danielperna84** [reply]
Also wenn ich jetzt einen HM-LC-Bl1PBU-FM runter fahren lasse, dann bekomme ich im Debug-Log:

```
...
Oct 10 19:05:10 hass2022 hass[176044]: 2022-10-10 19:05:10.445 DEBUG (MainThread) [hahomematic.xml_rpc_proxy] __async_request: ('setValue', ('OEQ0291353:1', 'LEVEL', 0.0))
Oct 10 19:05:10 hass2022 hass[176044]: 2022-10-10 19:05:10.643 DEBUG (MainThread) [hahomematic.client] _set_value: OEQ0291353:1, LEVEL, 0.0
Oct 10 19:05:10 hass2022 hass[176044]: 2022-10-10 19:05:10.644 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = OEQ0291353:1, parameter = LEVEL, value = 1.0
Oct 10 19:05:10 hass2022 hass[176044]: 2022-10-10 19:05:10.645 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = OEQ0291353:1, parameter = WORKING, value = True
Oct 10 19:05:10 hass2022 hass[176044]: 2022-10-10 19:05:10.645 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = OEQ0291353:1, parameter = DIRECTION, value = 2
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.147 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = CONTROL_MODE, value = 0
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = FAULT_REPORTING, value = 0
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = BATTERY_STATE, value = 2.7
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = VALVE_STATE, value = 10
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = BOOST_STATE, value = 0
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = ACTUAL_TEMPERATURE, value = 24.7
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = SET_TEMPERATURE, value = 24.0
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = PARTY_TEMPERATURE, value = 5.0
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = PARTY_START_TIME, value = 0
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = PARTY_START_DAY, value = 1
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = PARTY_START_MONTH, value = 1
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = PARTY_START_YEAR, value = 0
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = PARTY_STOP_TIME, value = 0
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = PARTY_STOP_DAY, value = 1
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = PARTY_STOP_MONTH, value = 1
Oct 10 19:05:11 hass2022 hass[176044]: 2022-10-10 19:05:11.152 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = NEQ1493025:4, parameter = PARTY_STOP_YEAR, value = 0
...
```

Und wenn ich stoppe:
```
...
Oct 10 19:05:25 hass2022 hass[176044]: 2022-10-10 19:05:25.570 DEBUG (MainThread) [hahomematic.xml_rpc_proxy] __async_request: ('setValue', ('OEQ0291353:1', 'STOP', True))
Oct 10 19:05:25 hass2022 hass[176044]: 2022-10-10 19:05:25.753 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = OEQ0291353:1, parameter = LEVEL, value = 0.37
Oct 10 19:05:25 hass2022 hass[176044]: 2022-10-10 19:05:25.754 DEBUG (XmlRpcServer on port 47369) [hahomematic.entity] event: hassbian-BidCos-RF, OEQ0291353:1, LEVEL
Oct 10 19:05:25 hass2022 hass[176044]: 2022-10-10 19:05:25.756 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = OEQ0291353:1, parameter = WORKING, value = False
Oct 10 19:05:25 hass2022 hass[176044]: 2022-10-10 19:05:25.756 DEBUG (XmlRpcServer on port 47369) [hahomematic.xml_rpc_server] event: interface_id = hassbian-BidCos-RF, channel_address = OEQ0291353:1, parameter = DIRECTION, value = 0
Oct 10 19:05:25 hass2022 hass[176044]: 2022-10-10 19:05:25.756 DEBUG (MainThread) [hahomematic.client] _set_value: OEQ0291353:1, STOP, True
...
```

In der Zwischenzeit kommt nix weiter für `WORKING`. Das ist also schon ziemlich synchron mit dem, was die Hardware macht.

**Reply by maxlin1** [reply]
@danielperna84  wie kann ich das Debug log auslesen? Ich bin noch relativ neu in homeassistatant? Dann kann ich bei mir auch mal schauen

**Reply by SukramJ** [reply]
WORKING könnte man separat nehmen.
DIRECTION sollte währen der Bewegung an der Entität als opening/closing angezeigt werden.

**Reply by maxlin1** [reply]
Working wäre ja gut nur siehe oben bei meiner Meldung den Screenshot in der Rule kann man es nicht verwenden, bzw. der Auslöser wird nicht ausgelöst.  Hast du das mal getestet @danielperna84

**Reply by danielperna84** [reply]
Also ich sehe im Logbuch nix zu opening / closing. Im Logbuch gibt es nur Einträge sobald der Rolladen offen (>0) oder geschlossen (==0) ist. Dazwischen passiert nichts. Zumindest nicht sichtbar für mich. 

<img width="473" alt="rsclip" src="https://user-images.githubusercontent.com/7396998/194938184-70d4391e-2e44-4ddd-9f11-8bb21d68d67b.png">

**Reply by SukramJ** [reply]
Ich glaub ich hab was bei hahomematic gefunden. Ich versuch das mal mit pydevccu zu testen.

**Reply by maxlin1** [reply]
genauso wie bei mir  und auch im Editor sehe ich kein working

**Comment by maxlin1** [comment]
<img width="1070" alt="image" src="https://user-images.githubusercontent.com/10881081/194939091-81b215a6-3ffe-4815-95c5-b660bec80a0e.png">
@danielperna84  geht dieser Auslöser in einer Regel bei dir ?

**Reply by danielperna84** [reply]
Ich habe keine Regel für sowas, und das auch nicht getestet. Warten wir mal ab was SukramJ bei seinen Tests herausfindet.

**Reply by SukramJ** [reply]
Wenn der Status nicht gesetzt wird kann auch dieser Auslöser nicht funktionieren.
Bitte warten

**Reply by maxlin1** [reply]
ah okay ja

**Reply by SukramJ** [reply]
Versucht mal die version 1.19.4.
Jetzt sollte der opening/closing Zwischenstatus angezeigt werden, und dann auch der Auslöser in der Automation funktionieren

**Reply by danielperna84** [reply]
Jupp. Auslöse in der Automation zum öffnen hat funktioniert. Und das Icon ändert sich jetzt auch. 👍

**Reply by maxlin1** [reply]
Jap funktioniert auch bei mir SUPER vielen Dank

**Reply by maxlin1** [reply]
Das der Falsche Prozent Wert angezeigt wird liegt denke ich jedoch an der RasberryMatic denn auch dort in der UI wird der Wert falsch angezeigt.

**Reply by SukramJ** [reply]
Wenn RM den gleichen Wert anzeigt …

**Comment by hvorragend** [comment]
Ich habe mich total gewundert, warum einige Automationen nicht mehr funktionieren und ich wusste auch nicht, wie lange das schon so ist. 

Ich prüfe halt den Wechsel von "from: open" zu "to: closed". Und plötzlich gibt es jeweils einen Zwischenstatus "wird geöffnet" und "wird geschlossen".  

Ist das eine HA-Neuerung oder kam das durch ein Update der Integration?

Bleibt das so? Dann lass ich mir einfach eine andere Logik einfallen.

**Reply by SukramJ** [reply]
Das bleibt so.
